### PR TITLE
Storing email template body as JSON.

### PR DESCRIPTION
**Pull request by mefarazath**
* $subject to overcome the limitation of storing the email template content (subject, body, footer) as a String joined with '|' .

* Some minor refactoring.

### CODE REVIEW

File: components/email-mgt/org.wso2.carbon.email.mgt/pom.xml [hunk 1]
```diff
@@ -113,6 +117,8 @@
 
                             org.wso2.carbon.stratos.common.*; version="${carbon.commons.imp.pkg.version}",
```
Comment (kasungayan) [inline components/email-mgt/org.wso2.carbon.email.mgt/pom.xml]: Remove unnecessary new line.

File: components/email-mgt/org.wso2.carbon.email.mgt/pom.xml [hunk 2]
```diff
@@ -113,6 +117,8 @@
...
 
+                            com.google.gson.*; version="${com.google.code.gson.osgi.version.range}",
+
```
Comment (kasungayan) [inline components/email-mgt/org.wso2.carbon.email.mgt/pom.xml]: Remove unnecessary new line.

File: components/email-mgt/org.wso2.carbon.email.mgt/pom.xml (hunk 1 above)
Comment (mefarazath) [inline components/email-mgt/org.wso2.carbon.email.mgt/pom.xml]: the new line was purely to keep the groups of bundle imports clear.

File: components/email-mgt/org.wso2.carbon.email.mgt/pom.xml (hunk 2 above)
Comment (mefarazath) [inline components/email-mgt/org.wso2.carbon.email.mgt/pom.xml]: the new line was purely to keep the groups of bundle imports clear.

File: components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/util/I18nEmailUtil.java [hunk 3]
```diff
@@ -203,35 +203,41 @@ public static Resource createTemplateResource(EmailTemplate emailTemplateDTO) th
...
-                    String errorMsg = "Template %s:%s contains '|' character which is invalid.";
+                String[] templateContentElements;
+                try {
+                    templateContentElements = new Gson().fromJson(templateContent, String[].class);
+                } catch (JsonSyntaxException ex) {
+                    String error = "Error deserializing '%s:%s' template from tenant registry.";
+                    throw new I18nEmailMgtServerException(String.format(error, templateDisplayName, locale), ex);
+                }
+
+                if (templateContentElements.length != 3) {
```
Comment (GayanM) [inline components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/util/I18nEmailUtil.java]: possible NPE if templateContentElements is null

Comment (mefarazath) [inline components/email-mgt/org.wso2.carbon.email.mgt/src/main/java/org/wso2/carbon/email/mgt/util/I18nEmailUtil.java]: fixed with https://github.com/wso2-extensions/identity-event-handler-notification/pull/29

**Pull request merged**